lib/b2c/sessions.ts

```diff
@@ -158,24 +575,30 @@ export class Sessions {
       options
     );
 
+    // The last_authenticated_at field from local authentication is still a string.
+    // Let's strongly type it now and then convert it to a Date.
+    type authFactorRow = Omit<AuthenticationFactor, "last_authenticated_at"> & {
+      last_authenticated_at: string;
+    };
+    const factorRows = sess.authentication_factors as authFactorRow[];
+    const typedFactors: AuthenticationFactor[] = [];
+    factorRows.forEach((factor) => {
+      typedFactors.push({
+        ...factor,
+        last_authenticated_at: new Date(factor.last_authenticated_at),
```

Comment:
As an aside - one thing that drives me nuts is that we convert sessions into `Date` objects here, but we _don't_ do the conversion on any other endpoints that return sessions (so, all Authenticate endpoints.) Since we're doing a major version bump, HDYF about dropping the date parsing entirely?

Reply:
Meant to respond to this earlier -- if I now annotate fields as `Date`, is there an easy way to extend the `request` function to parse these into new Date objects programmatically?